src/operator/src/statement/copy_database.rs

```diff
@@ -0,0 +1,253 @@
+            if table_name == "numbers" {
+                continue;
+            }
+            let mut table_file = req.location.clone();
+            table_file.push_str(&table_name);
+            table_file.push_str(suffix);
+            info!(
+                "Copy table: {}.{}.{} to {}",
+                req.catalog_name, req.schema_name, table_name, table_file
+            );
+
+            let exported = self
+                .copy_table_to(
+                    CopyTableRequest {
+                        catalog_name: req.catalog_name.clone(),
+                        schema_name: req.schema_name.clone(),
+                        table_name,
+                        location: table_file,
+                        with: req.with.clone(),
+                        connection: req.connection.clone(),
+                        pattern: None,
+                        direction: CopyDirection::Export,
+                        timestamp_range: req.time_range,
+                    },
+                    ctx.clone(),
+                )
+                .await?;
+            exported_rows += exported;
+        }
+        Ok(Output::AffectedRows(exported_rows))
+    }
+
+    /// Imports data to database from a given location and returns total rows imported.
+    #[tracing::instrument(skip_all)]
+    pub(crate) async fn copy_database_from(
+        &self,
+        req: CopyDatabaseRequest,
+        ctx: QueryContextRef,
+    ) -> error::Result<Output> {
+        // location must end with /
+        ensure!(
+            req.location.ends_with('/'),
+            InvalidCopyDatabasePathSnafu {
+                value: req.location,
+            }
+        );
+
+        info!(
+            "Copy database {}.{} from dir: {}, time: {:?}",
+            req.catalog_name, req.schema_name, req.location, req.time_range
+        );
+        let suffix = Format::try_from(&req.with)
+            .context(error::ParseFileFormatSnafu)?
+            .suffix();
+
+        let entries = list_files_to_copy(&req, suffix).await?;
+
+        let continue_on_error = req
+            .with
+            .get(CONTINUE_ON_ERROR_KEY)
+            .and_then(|v| bool::from_str(v).ok())
+            .unwrap_or(false);
+        let mut rows_inserted = 0;
+
+        for e in entries {
+            let table_name = match parse_file_name_to_copy(&e) {
+                Ok(table_name) => table_name,
+                Err(err) => {
+                    if continue_on_error {
+                        error!(err; "Failed to import table from file: {:?}", e);
+                        continue;
+                    } else {
+                        return Err(err);
+                    }
+                }
+            };
+            let req = CopyTableRequest {
+                catalog_name: req.catalog_name.clone(),
+                schema_name: req.schema_name.clone(),
+                table_name: table_name.clone(),
+                location: format!("{}/{}", req.location, e.path()),
```

Comment:
> Use join_path?

`join_path` does not apply here since it removes leading `/` in path:
https://github.com/GreptimeTeam/greptimedb/blob/75975adcb6a03dc68fa81e2cfe36f30a0a202538/src/object-store/src/util.rs#L81